src/patternfly/components/Chip/chip.scss

```diff
@@ -63,63 +52,39 @@
-    --pf-c-chip--BackgroundColor: var(--pf-c-chip--m-overflow--BackgroundColor);
-
-    // Remove chip border
-    &::before {
-      --pf-c-chip--BorderWidth: var(--pf-c-chip--m-overflow--BorderWidth);
-    }
-
-    .pf-c-button {
-      padding-right: var(--pf-c-chip--m-overflow--c-button--PaddingRight);
-      padding-left: var(--pf-c-chip--m-overflow--c-button--PaddingLeft);
-
-      &::after {
-        border: var(--pf-c-chip--m-overflow--c-button--BorderWidth) solid var(--pf-c-chip--m-overflow--c-button--active--BorderColor);
-        border-radius: var(--pf-c-chip--m-overflow--c-button--BorderRadius);
-      }
+    border: 0;
 
-      &.pf-m-hover,
-      &:hover {
-        // stylelint-disable-next-line
-        &::after {
-          --pf-c-chip--m-overflow--c-button--BorderWidth: var(--pf-c-chip--m-overflow--c-button--hover--BorderWidth);
-          --pf-c-chip--m-overflow--c-button--BorderColor: var(--pf-c-chip--m-overflow--c-button--hover--BorderColor);
-        }
+    &.pf-m-hover,
+    &:hover {
+      &::before {
+        --pf-c-chip--BorderColor: var(--pf-c-chip--m-overflow--hover--before--BorderColor);
       }
+    }
 
-      &.pf-m-active,
-      &:active {
-        // stylelint-disable-next-line
-        &::after {
-          --pf-c-chip--m-overflow--c-button--BorderWidth: var(--pf-c-chip--m-overflow--c-button--active--BorderWidth);
-          --pf-c-chip--m-overflow--c-button--BorderColor: var(--pf-c-chip--m-overflow--c-button--active--BorderColor);
-        }
+    &.pf-m-focus,
+    &:focus {
+      &::before {
+        --pf-c-chip--BorderColor: var(--pf-c-chip--m-overflow--focus--before--BorderColor);
       }
+    }
 
-      &.pf-m-focus,
-      &:focus {
-        // stylelint-disable-next-line
-        &::after {
-          --pf-c-chip--m-overflow--c-button--BorderWidth: var(--pf-c-chip--m-overflow--c-button--focus--BorderWidth);
-          --pf-c-chip--m-overflow--c-button--BorderColor: var(--pf-c-chip--m-overflow--c-button--focus--BorderColor);
-        }
+    &.pf-m-active,
+    &:active {
+      &::after {
+        --pf-c-chip--BorderColor: var(--pf-c-chip--m-overflow--active--before--BorderColor);
       }
     }
   }
 
-  &.pf-m-read-only {
-    padding-top: var(--pf-c-chip--m-read-only--PaddingTop);
-    padding-right: var(--pf-c-chip--m-read-only--PaddingRight);
-    padding-bottom: var(--pf-c-chip--m-read-only--PaddingBottom);
-  }
-
   // Button
   .pf-c-button {
-    padding-right: var(--pf-c-chip--c-button--PaddingRight);
-    padding-left: var(--pf-c-chip--c-button--PaddingLeft);
+    margin-top: var(--pf-c-chip--c-button--MarginTop);
+    margin-right: var(--pf-c-chip--c-button--MarginRight);
+    margin-bottom: var(--pf-c-chip--c-button--MarginBottom);
     font-size: var(--pf-c-chip--c-button--FontSize);
+
+    --pf-c-button--PaddingRight: var(--pf-c-chip--c-button--PaddingRight);
+    --pf-c-button--PaddingLeft: var(--pf-c-chip--c-button--PaddingLeft);
   }
```

Comment:
I wonder if we should add `position: relative` to the `.pf-c-chip__text` element so that you can interact with it (like select the text). We could also achieve the border effect with an actual `border` instead of a pseudo element (which would allow you to interact with the text), though I think the pseudo element is a little more flexible, and makes it easier to make future changes like changing the border width on hover. WDYT? If we want to change to a `border`, now would be good to do it. If not, we could add `position` or sort that out later as the current chip is has this issue so it's not pressing.

Reply:
Also can we make sure the border vars are named from what they apply to? Currently they apply to `::before` but don't reference that in the name.

Reply:
I think we should keep the pseudo element for the border because i think its reasonable that design might ask for the border width to increase on H/A/F. So I just added `position: relative` to the text but you can only highlight text in the `read-only` chip now.. also updated the name